packages/appstore/src/components/wallet-transfer/wallet-transfer-tile/wallet-transfer-tile.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Text, AppLinkedWithWalletIcon, WalletIcon } from '@deriv/components';
+import { formatMoney } from '@deriv/shared';
+import { localize } from '@deriv/translations';
+import { getAccountName } from 'Constants/utils';
+import { WalletJurisdictionBadge } from 'Components/wallet-jurisdiction-badge';
+import type { TTransferAccount } from 'Types';
+import './wallet-transfer-tile.scss';
+
+type TIconSize =
+    | React.ComponentProps<typeof AppLinkedWithWalletIcon>['size']
+    | React.ComponentProps<typeof WalletIcon>['size'];
+
+type TWalletTileProps = {
+    account?: TTransferAccount;
+    className?: string;
+    has_hover?: boolean;
+    icon_size?: TIconSize;
+    is_active?: boolean;
+    is_list_item?: boolean;
+    is_mobile?: boolean;
+    onClick?: () => void;
+};
+
+const WalletTransferTile = ({
+    account,
+    className,
+    has_hover,
+    icon_size = 'small',
+    is_active,
+    is_list_item,
+    is_mobile,
+    onClick,
+}: TWalletTileProps) => {
+    const IconComponent = React.useCallback(() => {
+        if (account?.account_type === 'wallet') {
+            return account?.icon ? (
+                <WalletIcon
+                    gradient_class={account?.gradient_class}
+                    icon={account?.icon}
+                    size={icon_size as React.ComponentProps<typeof WalletIcon>['size']}
+                    type={account?.type}
+                />
+            ) : null;
+        }
+
+        return account?.icon && account?.active_wallet_icon ? (
+            <AppLinkedWithWalletIcon
+                app_icon={account?.icon}
+                gradient_class={account?.gradient_class || ''}
+                size={icon_size as React.ComponentProps<typeof AppLinkedWithWalletIcon>['size']}
+                type={account?.type}
+                wallet_icon={account?.active_wallet_icon}
+            />
+        ) : null;
+    }, [
+        account?.account_type,
+        account?.active_wallet_icon,
+        account?.gradient_class,
+        account?.icon,
+        account?.type,
+        icon_size,
+    ]);
+
+    const Label = React.useCallback(() => {
+        let size;
+        if (is_list_item) size = is_mobile ? 'xxs' : 'xs';
+        else size = is_mobile ? 'xxxxs' : 'xxxs';
+
+        return (
+            <Text as='div' size={size} weight='bold'>
+                {getAccountName({ ...account })}
+            </Text>
+        );
+    }, [account, is_mobile, is_list_item]);
```

Comment:
```suggestion
    const Label = React.useCallback(() => {
        let size;
        if (is_list_item) size = is_mobile ? 'xxs' : 'xs';
        else size = is_mobile ? 'xxxxs' : 'xxxs';

        return (
            <Text as='div' size={size} weight='bold'>
                {getAccountName({ ...account })}
            </Text>
        );
    }, [account, is_mobile, is_list_item]);
```
Please split this component out instead of doing this. Yes this solves re-rendering but useCallback has its own overhead and re-renders can be avoid by moving it out of this component into its own. It is also cleaner and more readable this way. Do it for the ones i mark with "Please split"